### ISSUE TITLE
Add page grid view

### ISSUE THREAD

**Issue by barklund** [ampproject/amp-wp#3836]
## Feature description

Add a page grid view as a new global state for the entire editor as well as a button in the page carousel that enables it.

In grid view, show page outlines of all pages in story and allow them to be both dragged and cut-pasted as designed.

Page reordering/duplication takes immediate effect.

Page outlines can be resized using the slider in the footer.

The close button in the footer will close grid view (changes have been persisted instantly when made).

This ticket does not include showing actual previews of the pages in the grid view.

The keyboard interactions are dependent on the outcome of #3826.

This is still pending UX review.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Users can enter grid view when pressing the button in the footer.
* Users can reorder pages in grid view by dragging.
* Users can focus pages and spaces in grid view by clicking.
* Users can move pages in grid view by using keyboard shortcuts as designed.
* Users can resize pages in grid view.
* Users can exist grid view.

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by jauyong** [comment]
Hey @barklund what exactly is this AC? Are you talking about the zoom slider?
> Users can resize pages in grid view.

**Comment by barklund** [comment]
> Hey @barklund what exactly is this AC? Are you talking about the zoom slider?
> 
> > Users can resize pages in grid view.

Yes, it was referring to the preview thumbnail sizing. I have updated and clarified the text (and added _deleting pages_).

Note that the AC does not include multi-select as we're still exploring if this is a P1 or later.

**Comment by adamsilverstein** [comment]
I love this issue, having a grid mode will make layout control so much easier for users! 

One idea from a recent demo: how about enabling the grid mode automatically (or a "grid automatic mode") for specific interactions such as drag-and-drop or image resizing. In this mode, the grid isn't visible until until interaction - when a user starts resizing or dragging image for example, then the grid appears and interactions "snap" to it. Once the stop interacting, the grid disappears.

**Comment by swissspidy** [comment]
@adamsilverstein I think this issue is more about a grid view as described in https://github.com/ampproject/amp-wp/issues/3974#issuecomment-572136169. Snapping is handled in ampproject/amp-wp#3773.

**Comment by adamsilverstein** [comment]
@swissspidy - thanks for the link, that issue does cover my idea - I had missed it earlier when reviewing the issue backlog.

**Comment by swissspidy** [comment]
**Two open UX questions:**

* Should clicking on the rectangles at the top change the zoom value? Or are they purely decoration?
* How many zoom levels should there be? Currently there are three steps: small, medium (default), large. We could add more, or even make it totally fluid/responsive.

Current implementation:

![Screenshot 2020-01-22 at 14 35 35](https://user-images.githubusercontent.com/841956/72898667-d337fc00-3d24-11ea-8c6a-4725ddef5aee.png)
![Screenshot 2020-01-22 at 14 35 46](https://user-images.githubusercontent.com/841956/72898668-d337fc00-3d24-11ea-8626-9f5b191781a1.png)
![Screenshot 2020-01-22 at 14 35 57](https://user-images.githubusercontent.com/841956/72898669-d3d09280-3d24-11ea-8508-bbe2d98d9ec0.png)

**Comment by samitron7** [comment]
@swissspidy 

Here are answers to the UX question:
1. They are purely decoration
2. I think 3 zoom level is fine for now. Can you give me the sizes for the smallest and largest thumbnail.
